File: src/main/java/com/university/services/impl/DepartmentServiceImpl.java
package com.university.services.impl;

import com.university.entities.Department;
import com.university.services.DepartmentService;
import org.springframework.beans.factory.annotation.Autowired;
import com.university.repositories.DepartmentRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class DepartmentServiceImpl implements DepartmentService {

    private final DepartmentRepository departmentRepository;

    @Autowired
    public DepartmentServiceImpl(DepartmentRepository departmentRepository) {
        this.departmentRepository = departmentRepository;
    }

    @Override
    public void addDepartment(Department department) {
        departmentRepository.saveAndFlush(department);
    }

    @Override
    public void removeDepartment(int id) {
        departmentRepository.delete(id);
    }

    @Override
    public Department getDepartment(int id) {
        return departmentRepository.findOne(id);
    }

    @Override
    public List<Department> getAllDepartments() {
        return departmentRepository.findAllByOrderByNameAsc();
    }
}

File: src/main/kotlin/com/university/services/impl/DepartmentServiceImpl.kt
package com.university.services.impl

import com.university.entities.Department
import com.university.repositories.DepartmentRepository
import com.university.services.DepartmentService
import org.springframework.stereotype.Service

@Service
class DepartmentServiceImpl (private val departmentRepository: DepartmentRepository) : DepartmentService {
    override fun getAll(): List<Department> {
        return departmentRepository.findAllByOrderByNameAsc()
    }
}
